data/src/history/manager.rs

```diff
@@ -499,6 +515,62 @@ impl Data {
         })
     }
 
+    fn history_banner_view(
+        &self,
+        server: &server::Server,
+        kind: &history::Kind,
+    ) -> Option<history::BannerView> {
+        let History::Full { messages, .. } = self.map.get(server)?.get(kind)? else {
+            return None;
+        };
```

Comment:
I don't think we should rely on our history to extract the topic at all. 

When we join a channel, which happens every time we start up, we should always receive RPL_TOPIC from the server. I think we can just hold a map of latest topic messages per channel in our client state and update it when those messages come in.

Then we can can query the topic from our client state vs dealing with history.

We can then remove all the metadata from message we've added here except for a single new server kind for Topic so we can filter those messages out if the banner option is chosen.

I think this will simplify a lot. Let me know if you have any questions or if you'd prefer I implemented it, but if you want to give it a shot, let me know!

Reply:
Sounds like a good idea to me.  I will have to learn more about how commands/messages get passed around in iced/Halloy, but that's something I want to learn so I'd like to give it a shot.  I will likely have questions for you, but will see what I can figure out on my own first.

Reply:
Great! You can start here in data::client where we handle all messages:

https://github.com/squidowl/halloy/blob/cb6c624c77ef363df2068fa4c9002148b427dbe5/data/src/client.rs#L182

You can setup some new state on Client, update it when Topic messages come in, then expose a method to get the current topic for a server / channel and call it from view!

Reply:
This was a *very* helpful set of pointers, thank you!  I have made a first attempt at the recommended change.  I think it's quite a bit clearer/simpler and it seems to be working well here.

Reply:
That is much better! Glad it was helpful.

I've pushed a few changes to get this across the finish line, let me know what you think.

@casperstorm LGTM from a technical standpoint, you can do a final design review

Reply:
Looks good to me.  Very instructive, thank you :)

The only "change" I have is a reversion of a change I made previously.  I had removed a `vertical_space` from before the `input_view` when switching to `spacing` because I couldn't figure out a clean way to include the same padding to the topic banner.  You found a clean way to do so for the topic banner and removed the `spacing` appropriately, but that meant there wasn't any spacing between the `input_view` and the `history_view` anymore.  So I put that back.

Reply:
Can we add that same space to the other buffers then so we're consistent?

Reply:
Man I'm sorry, total brain fart from me on this one.

Reply:
Happens to the best of us :)